src/OpenTracing.MockTracer/MockTracer.cs

```diff
@@ -0,0 +1,225 @@
+                {
+                    return new MockSpan.MockContext(traceId.Value, spanId.Value, baggage);
+                }
+
+                return null;
+            }
+        }
+
+        public sealed class MockSpanBuilder : ISpanBuilder
+        {
+            private readonly MockTracer _tracer;
+            private readonly string _operationName;
+            private DateTimeOffset _startTime = DateTimeOffset.MinValue;
+            private readonly List<MockSpan.Reference> _spanReferences = new List<MockSpan.Reference>();
+            private readonly Dictionary<string, object> _initialTags = new Dictionary<string, object>();
+
+            public MockSpanBuilder(MockTracer tracer, string operationName)
+            {
+                _tracer = tracer;
+                _operationName = operationName;
+            }
+
+            public ISpanBuilder AsChildOf(ISpan parent)
+            {
+                if (parent == null) 
+                    return this;
+                return AsChildOf(parent.Context);
+            }
+
+            public ISpanBuilder AsChildOf(ISpanContext parent)
+            {
+                if (parent == null)
+                    return this;
+                return AddReference(References.ChildOf, parent);
+            }
+
+            public ISpanBuilder FollowsFrom(ISpan parent)
+            {
+                return FollowsFrom(parent.Context);
+            }
+
+            public ISpanBuilder FollowsFrom(ISpanContext parent)
+            {
+                return AddReference(References.FollowsFrom, parent);
+            }
+
+            public ISpanBuilder AddReference(string referenceType, ISpanContext referencedContext)
+            {
+                if (referencedContext != null)
+                {
+                    _spanReferences.Add(new MockSpan.Reference((MockSpan.MockContext)referencedContext, referenceType));
+                }
+
+                return this;
+            }
+
+            public ISpanBuilder WithTag(string key, bool value)
+            {
+                _initialTags[key] = value;
+                return this;
+            }
+
+            public ISpanBuilder WithTag(string key, double value)
+            {
+                _initialTags[key] = value;
+                return this;
+            }
+
+            public ISpanBuilder WithTag(string key, int value)
+            {
+                _initialTags[key] = value;
+                return this;
+            }
+
+            public ISpanBuilder WithTag(string key, string value)
+            {
+                _initialTags[key] = value;
+                return this;
+            }
+
+            public ISpanBuilder WithStartTimestamp(DateTimeOffset startTimestamp)
+            {
+                _startTime = startTimestamp;
+                return this;
+            }
+
+            public ISpan Start()
+            {
+                if (_startTime == DateTimeOffset.MinValue) // value was not set by builder
```

Comment:
Probably just pending merge with #50, but this does not handle `ITracer.ActiveSpan`